### PR TITLE
api docs: Add descriptions for Scaffold, Theme, and ThemeData classes.

### PR DESCRIPTION
**Pull request by qchong**

### CODE REVIEW

**Comment by qchong** [comment]
@abarth @Hixie Can you please review?

**Comment by qchong** [comment]
Fix for flutter/flutter/issues/2541

File: packages/flutter/lib/src/material/theme.dart [hunk 1]
```diff
@@ -20,6 +21,7 @@ class Theme extends InheritedWidget {
     assert(data != null);
   }
 
+/// Holds attribute values that specify the look and format of this widget.
```
Comment (abarth) [inline packages/flutter/lib/src/material/theme.dart]: Bad indent.  Also, I'd skip the introductory phrasing.  Perhaps:  "Specifies the color and typography for descendant widgets."

File: packages/flutter/lib/src/material/theme_data.dart [hunk 2]
```diff
@@ -27,6 +27,9 @@ const Color _kLightThemeSplashColor = const Color(0x66C8C8C8);
 const Color _kDarkThemeHighlightColor = const Color(0x40CCCCCC);
 const Color _kDarkThemeSplashColor = const Color(0x40CCCCCC);
 
+/// Holds the current attribute values for a particular theme.
```
Comment (abarth) [inline packages/flutter/lib/src/material/theme_data.dart]: I'd be more specific about what sort of attributes.  For example, "Color and typography values for a material design theme."

File: packages/flutter/lib/src/material/scaffold.dart [hunk 3]
```diff
@@ -175,6 +175,7 @@ class _FloatingActionButtonTransitionState extends State<_FloatingActionButtonTr
   }
 }
 
+/// Component with a material design-style [visual structure](https://www.google.com/design/spec/layout/structure.html) suitable for mobile apps.
```
Comment (Hixie) [inline packages/flutter/lib/src/material/scaffold.dart]: Here's a possible alternative description for this component:
```
/// Implements the basic material design visual layout structure, and provides APIs for
/// showing drawers, snackbars, and bottom sheets.
///
/// See: https://www.google.com/design/spec/layout/structure.html
```

File: packages/flutter/lib/src/material/theme_data.dart (hunk 2 above)
Comment (qchong) [inline packages/flutter/lib/src/material/theme_data.dart]: Thanks, fixed in latest push.

File: packages/flutter/lib/src/material/theme.dart (hunk 1 above)
Comment (qchong) [inline packages/flutter/lib/src/material/theme.dart]: Thanks, fixed in latest push.

File: packages/flutter/lib/src/material/theme.dart [hunk 4]
```diff
@@ -10,6 +10,7 @@ export 'theme_data.dart' show ThemeData, ThemeBrightness;
 
 const kThemeAnimationDuration = const Duration(milliseconds: 200);
 
+/// Defines a widget that is configured with a theme.
```
Comment (qchong) [inline packages/flutter/lib/src/material/theme.dart]: Thanks, fixed in latest push.

File: packages/flutter/lib/src/material/scaffold.dart (hunk 3 above)
Comment (qchong) [inline packages/flutter/lib/src/material/scaffold.dart]: Thanks, fixed in latest push. I've split the first sentence to keep it brief.

**Comment by abarth** [comment]
LGTM

**Pull request merged**